packages/react/src/components/Authenticator/hooks/useFormHandlers/useFormHandlers.ts

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { useCallback } from 'react';
+import { getFormDataFromEvent } from '../../../../helpers/utils';
+import { useAuthenticator } from '../useAuthenticator';
+
+export default function useFormHandlers() {
+  const { submitForm, updateBlur, updateForm } = useAuthenticator();
+
+  const handleBlur = useCallback(
+    ({ target: { name } }: React.FocusEvent<HTMLFormElement>) => {
+      updateBlur({ name });
+    },
+    [updateBlur]
+  );
+
+  const handleChange = useCallback(
+    ({ target: { name, value } }: React.ChangeEvent<HTMLFormElement>) => {
+      updateForm({ name, value });
+    },
```

Comment:
This looks good!
Just one quick update. We still need to do the check here if it's a checkbox and set the value undefined. Otherwise it will break the `custom-sign-up-fields` example. We should also add another test for this use case as well. 

It occurs when you toggle the checkbox. It works the first time, but not the second time you toggle it. 

https://user-images.githubusercontent.com/65630/159376814-92d8e9e7-d13d-4295-bd04-5b81990b147d.mp4

Reply:
Ahhh, thanks for the heads up!